src/engine/shared.rs

```diff
@@ -196,6 +198,13 @@ pub fn validate_name(name: &str) -> StratisResult<()> {
             format!("Name contains control characters : {}", name),
         ));
     }
+    let name_udevregex = Regex::new(r"[\$!]").unwrap();
```

Comment:
You can change the unwrap() into an expect(). If it is a valid regular expression, then the function must succeed, and you can guarantee that ti's a valid regular expression by looking at it.

Reply:
You can also follow the advice of the regex maintainers about using lazy_static(), I should think. The instructions are here: https://docs.rs/regex/1.4.0/regex/index.html#example-avoid-compiling-the-same-regex-in-a-loop. There was no point in using lazy_static in stratis_uuids_to_names, because the regular expression is used only once, but since stratisd is a long running program it's definitely worth it here.

Reply:
OK; I just changed it to an expect() call.